src/Microsoft.Health.Fhir.Liquid.Converter.UnitTests/Filters/DateFiltersTests.cs

```diff
@@ -50,7 +50,7 @@ public static IEnumerable<object[]> GetValidDataForAddSeconds()
             yield return new object[] { string.Empty, 60, "local", string.Empty };
             yield return new object[] { @"1970-01-01T00:01:00.000+10:00", -60, "utc", @"1969-12-31T14:00:00.000Z" };
             yield return new object[] { @"1970-01-01T00:01:00.000+05:30", -60, "utc", @"1969-12-31T18:30:00.000Z" };
-            yield return new object[] { @"1970-01-01T00:01:00Z", 60.123, "preserve", @"1970-01-01T00:02:00.123Z" };
+            yield return new object[] { @"1970-01-01T00:01:00Z", 60.123, "preserve", @"1970-01-01T00:02:00.122Z" };
```

Comment:
Why was this change made?

Reply:
It has changed to resolve one of the unit test failing

Reply:
Sorry, I missed this earlier. This doesn't look correct.  It seems like with the upgrade to .NET verisons the behavior has changed.  Can you please dig deeper to see if there is a way to preserve the original?

Reply:
I did some investigations and looks like the value of date time which is calculated for adding 60.123 seconds is {00:02:00.1229999} time of the day, number of ticks is also calculated : 621355969201229999 , I think that's why in test it is expecting 1970-01-01T00:02:00.122Z.
if we want to have  a unit test which both the input and result having same seconds we can use 60.124 , since for 60.124 , the value of the daytime would be something like {00:02:00.1240000}

Reply:
So I dug a little deeper, this test is calling into:
https://github.com/microsoft/FHIR-Converter/blob/931a935a5bb3b9ccde4b1e9cf7396f2d94b95643/src/Microsoft.Health.Fhir.Liquid.Converter/Filters/DateFilters.cs#L17

which in turn is invoking:
https://github.com/microsoft/FHIR-Converter/blob/931a935a5bb3b9ccde4b1e9cf7396f2d94b95643/src/Microsoft.Health.Fhir.Liquid.Converter/Models/PartialDateTime.cs#L85

60.123 is supposed to represent 60 seconds and 123 miliiseconds so the expected output should be a minute greater and 123 milliseconds.  

I took a look at breaking changes in the .NET versions, I didn't see anything relevant in [.NET 8 ](https://learn.microsoft.com/en-us/dotnet/core/compatibility/8.0)
but I did see this in the .NET 7 breaking changes:
https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/7.0/datetime-add-precision

Digging deeper, if you use a value like 1.123 you get the expected number of milliseconds.  The value seems to deviate once you get to a second value of 8+.

To preserve the original logic I think we can modify the code in https://github.com/microsoft/FHIR-Converter/blob/931a935a5bb3b9ccde4b1e9cf7396f2d94b95643/src/Microsoft.Health.Fhir.Liquid.Converter/Models/PartialDateTime.cs#L85

from
```
DateTimeValue = DateTimeValue.AddSeconds(seconds);
```
to 
```
var timespan = TimeSpan.FromSeconds(seconds)
DateTimeValue = DateTimeValue.Add(timespan);
```

This seems to preseve the original intent.